### PR TITLE
[DR-2213] Add Mixpanel events

### PR DESCRIPTION
**Pull request by tcjiang9**
Add events for interactables in the catalogue ui

### CODE REVIEW

File: src/pages/library/common.js [hunk 1]
```diff
@@ -177,6 +179,12 @@ export const SearchAndFilterComponent = ({ fullList, sidebarSections, customSort
       filter: filter || undefined
     }, { addQueryPrefix: true })
 
+    if (filter) {
+      Ajax().Metrics.captureEvent(Events.catalogueFilter + ':search', {
+        filter: filter
+      })
```
Comment (dmohs) [inline src/pages/library/common.js]: JavaScript can give you a nice bit of concision here:
```suggestion
      Ajax().Metrics.captureEvent(Events.catalogueFilter + ':search', { filter })
```

**Review by dmohs — APPROVED** [review]

File: src/pages/library/common.js (hunk 1 above)
Comment (kyuksel) [reply src/pages/library/common.js]: The entirety of the `if` block could be even more concise 😊 

```javascript
!!filter && Ajax().Metrics.captureEvent(`${Events.catalogueFilter}:search`, { filter })
```

Comment (tcjiang9) [reply src/pages/library/common.js]: @kyuksel I think one-lining reduces clarity a bit too much

Comment (kyuksel) [reply src/pages/library/common.js]: `!!something && …` is a pattern we use a lot in this codebase so my brain actually parses it faster that way but I'm fine with making only the change @dmohs suggested and leaving it at

```javascript
if (filter) {
  Ajax().Metrics.captureEvent(`${Events.catalogueFilter}:search`, { filter })
}
```

if that reads much better for you.

File: src/libs/events.js [hunk 2]
```diff
@@ -20,6 +20,9 @@ const eventsList = {
   cloudEnvironmentCreate: 'cloudEnvironment:create',
   cloudEnvironmentDelete: 'cloudEnvironment:delete',
   cloudEnvironmentUpdate: 'cloudEnvironment:update',
+  catalogueFilter: 'catalogue:filter',
```
Comment (lu-c) [inline src/libs/events.js]: We've been using "Catalog" instead of "Catalogue" in the ui, can we keep that consistent by changing these?

**Comment by kyuksel** [comment]
@tcjiang9 Could you please address [this comment](https://github.com/DataBiosphere/terra-ui/pull/2677#discussion_r742379624), and rebase with `dev` and resolve conflicts?

**Comment by kyuksel** [comment]
@tcjiang9 There's been no activity on this PR for about a week now. Could you please close it if you're not planning on working on it soon?

**Comment by tcjiang9** [comment]
@kyuksel was blocked from deploying by integration test failure in dev, I thought I had addressed your comment, but I can move {filter} onto the same line

**Comment by kyuksel** [comment]
> was blocked from deploying by integration test failure in dev, I thought I had addressed your comment, but I can move {filter} onto the same line

@tcjiang9 Thank you. Could you please also rebase your PR branch with the `dev` branch?

**Review by kyuksel — APPROVED** [review]

**Pull request merged**